<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Final Project - Fluids</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="GLPanel" name="GLWidget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>731</width>
      <height>731</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
   </widget>
   <widget class="QFrame" name="parameterFrame">
    <property name="geometry">
     <rect>
      <x>749</x>
      <y>-1</y>
      <width>441</width>
      <height>731</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>431</width>
       <height>731</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Simulation Variables</string>
        </property>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="geometry">
          <rect>
           <x>9</x>
           <y>29</y>
           <width>411</width>
           <height>121</height>
          </rect>
         </property>
         <property name="title">
          <string>Addition Selector</string>
         </property>
         <widget class="QRadioButton" name="addVelocityRadio">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>30</y>
            <width>116</width>
            <height>22</height>
           </rect>
          </property>
          <property name="text">
           <string>Velocity</string>
          </property>
         </widget>
         <widget class="QRadioButton" name="addDensityRadio">
          <property name="geometry">
           <rect>
            <x>20</x>
            <y>50</y>
            <width>116</width>
            <height>22</height>
           </rect>
          </property>
          <property name="text">
           <string>Density</string>
          </property>
         </widget>
         <widget class="QLabel" name="label_14">
          <property name="geometry">
           <rect>
            <x>173</x>
            <y>11</y>
            <width>111</width>
            <height>17</height>
           </rect>
          </property>
          <property name="text">
           <string>Density Radius :</string>
          </property>
         </widget>
         <widget class="QLineEdit" name="densityRadiusEdit">
          <property name="geometry">
           <rect>
            <x>288</x>
            <y>7</y>
            <width>81</width>
            <height>27</height>
           </rect>
          </property>
         </widget>
         <widget class="QLabel" name="label_15">
          <property name="geometry">
           <rect>
            <x>173</x>
            <y>39</y>
            <width>111</width>
            <height>17</height>
           </rect>
          </property>
          <property name="text">
           <string>Velocity Radius :</string>
          </property>
         </widget>
         <widget class="QLineEdit" name="velocityRadiusEdit">
          <property name="geometry">
           <rect>
            <x>288</x>
            <y>35</y>
            <width>81</width>
            <height>27</height>
           </rect>
          </property>
         </widget>
        </widget>
        <widget class="QLabel" name="label">
         <property name="geometry">
          <rect>
           <x>90</x>
           <y>360</y>
           <width>151</width>
           <height>17</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Mass Spring System</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_3">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>580</y>
           <width>111</width>
           <height>17</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Rope Settings</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_4">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>430</y>
           <width>111</width>
           <height>17</height>
          </rect>
         </property>
         <property name="font">
          <font>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Rod Settings</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_5">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>400</y>
           <width>66</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Mass :</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_6">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>430</y>
           <width>141</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Max Connector Dist :</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_7">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>460</y>
           <width>66</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Density :</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_8">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>520</y>
           <width>66</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Bend K :</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_9">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>490</y>
           <width>71</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Stretch K :</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_10">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>550</y>
           <width>81</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Segments :</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="rodDensityEdit">
         <property name="geometry">
          <rect>
           <x>270</x>
           <y>460</y>
           <width>81</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="rodStretchEdit">
         <property name="geometry">
          <rect>
           <x>270</x>
           <y>490</y>
           <width>81</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="rodBendEdit">
         <property name="geometry">
          <rect>
           <x>270</x>
           <y>520</y>
           <width>81</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="rodSegmentsEdit">
         <property name="geometry">
          <rect>
           <x>270</x>
           <y>550</y>
           <width>81</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="ropeDensityEdit">
         <property name="geometry">
          <rect>
           <x>271</x>
           <y>606</y>
           <width>81</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="ropeBendEdit">
         <property name="geometry">
          <rect>
           <x>271</x>
           <y>636</y>
           <width>81</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="ropeSegmentsEdit">
         <property name="geometry">
          <rect>
           <x>271</x>
           <y>666</y>
           <width>81</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_11">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>640</y>
           <width>71</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Bend K :</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_12">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>610</y>
           <width>66</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Density :</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_13">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>670</y>
           <width>81</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Segments :</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="massEdit">
         <property name="geometry">
          <rect>
           <x>60</x>
           <y>400</y>
           <width>81</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="maxSpringDistEdit">
         <property name="geometry">
          <rect>
           <x>60</x>
           <y>450</y>
           <width>81</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_16">
         <property name="geometry">
          <rect>
           <x>5</x>
           <y>526</y>
           <width>81</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Max Strain :</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="springStiffnessEdit">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>492</y>
           <width>50</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_17">
         <property name="geometry">
          <rect>
           <x>5</x>
           <y>496</y>
           <width>101</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Base Stiffness:</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="maxStrainEdit">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>522</y>
           <width>50</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLineEdit" name="dampingStiffnessEdit">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>552</y>
           <width>51</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_18">
         <property name="geometry">
          <rect>
           <x>7</x>
           <y>555</y>
           <width>111</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Damp Stiffness :</string>
         </property>
        </widget>
        <widget class="QCheckBox" name="isFixedCheckBox">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>390</y>
           <width>97</width>
           <height>22</height>
          </rect>
         </property>
         <property name="text">
          <string>Is Fixed</string>
         </property>
        </widget>
        <widget class="QPushButton" name="startSimulationButton">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>210</y>
           <width>181</width>
           <height>27</height>
          </rect>
         </property>
         <property name="text">
          <string>Start Simulation</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_19">
         <property name="geometry">
          <rect>
           <x>149</x>
           <y>122</y>
           <width>141</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Velocity Magnitude :</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="velocityMagnitudeEdit">
         <property name="geometry">
          <rect>
           <x>297</x>
           <y>119</y>
           <width>81</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_20">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>95</y>
           <width>141</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Density Magnitude :</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="densityMagnitudeEdit">
         <property name="geometry">
          <rect>
           <x>297</x>
           <y>92</y>
           <width>81</width>
           <height>27</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLabel" name="timeStepLabel">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>250</y>
           <width>81</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Time Step:</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="timeStepEdit">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>250</y>
           <width>51</width>
           <height>21</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="timeStepLabel_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>280</y>
           <width>141</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Newton Tolerance :</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="newtonTolEdit">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>280</y>
           <width>51</width>
           <height>21</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="timeStepLabel_3">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>310</y>
           <width>131</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Newton Max Iters :</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="newtonMaxItersEdit">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>310</y>
           <width>51</width>
           <height>21</height>
          </rect>
         </property>
        </widget>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>580</y>
           <width>151</width>
           <height>121</height>
          </rect>
         </property>
         <property name="title">
          <string>Connector Type</string>
         </property>
         <widget class="QRadioButton" name="ropeButton">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>90</y>
            <width>116</width>
            <height>22</height>
           </rect>
          </property>
          <property name="text">
           <string>Rope</string>
          </property>
         </widget>
         <widget class="QRadioButton" name="springButton">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>30</y>
            <width>116</width>
            <height>22</height>
           </rect>
          </property>
          <property name="text">
           <string>Spring</string>
          </property>
         </widget>
         <widget class="QRadioButton" name="flexibleRodButton">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>70</y>
            <width>116</width>
            <height>22</height>
           </rect>
          </property>
          <property name="text">
           <string>Flexible Rod</string>
          </property>
         </widget>
         <widget class="QRadioButton" name="rigidRodButton">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>50</y>
            <width>116</width>
            <height>22</height>
           </rect>
          </property>
          <property name="text">
           <string>Rigid Rod</string>
          </property>
         </widget>
        </widget>
        <widget class="QLabel" name="label_21">
         <property name="geometry">
          <rect>
           <x>149</x>
           <y>149</y>
           <width>141</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Viscosity K:</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="viscosityConstantEdit">
         <property name="geometry">
          <rect>
           <x>298</x>
           <y>146</y>
           <width>81</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_22">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>177</y>
           <width>141</width>
           <height>17</height>
          </rect>
         </property>
         <property name="text">
          <string>Diffusion K:</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="diffusionConstantEdit">
         <property name="geometry">
          <rect>
           <x>298</x>
           <y>174</y>
           <width>81</width>
           <height>27</height>
          </rect>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuScene">
    <property name="title">
     <string>Scene</string>
    </property>
    <addaction name="actionReset"/>
    <addaction name="actionReset_Everything"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuScene"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="text">
    <string>Clear Scene</string>
   </property>
  </action>
  <action name="actionReset_Everything">
   <property name="text">
    <string>Reset Everything</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLPanel</class>
   <extends>QWidget</extends>
   <header>glpanel.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
